pkg/controller/fileobserver/observer_polling.go

```diff
@@ -74,28 +86,42 @@ func (o *pollingObserver) processReactors(stopCh <-chan struct{}) {
 		o.reactorsMutex.RLock()
 		defer o.reactorsMutex.RUnlock()
 		for filename, reactors := range o.reactors {
-			currentHash, err := calculateFileHash(filename)
-			if err != nil {
+			currentFileState, err := calculateFileHash(filename)
+			if err != nil && !os.IsNotExist(err) {
 				return false, err
 			}
-			lastKnownHash := o.files[filename]
 
-			// No file change detected
-			if lastKnownHash == currentHash {
-				continue
-			}
+			lastKnownFileState := o.files[filename]
+			o.files[filename] = currentFileState
 
-			klog.Infof("Observed change: file:%s (current: %q, lastKnown: %q)", filename, currentHash, lastKnownHash)
-			o.files[filename] = currentHash
+			klog.Infof("Observed change: file:%s (current: %q, lastKnown: %q)", filename, currentFileState.hash, lastKnownFileState.hash)
 
 			for i := range reactors {
-				action := FileModified
-				switch {
-				case len(lastKnownHash) == 0:
+				action := NoAction
+
+				// do nothing if file does not exists
+				if !lastKnownFileState.exists && !currentFileState.exists {
```

Comment:
all these cases would make a nice switch statement (again).

Reply:
@sttts i found the switch more confusing, but I can follow up? :-) at least I put comments on what each case mean